main.js

```diff
@@ -434,8 +434,14 @@ async function get_problems() {
 
     return false;
   }
-
   let final_problems = available_problems.slice(0, problems_cnt);
```

Comment:
Shouldn't you slice the `available_problems` from the first index to the minimum of the `problems_cnt` and `available_problems.length`?
Because, as far as I know, the function `slice` would throw an exception if you gave it an index that is out of bounds.

Reply:
I’ve made the changes you requested and pushed the updates to the PR. Please let me know if there’s anything else that needs to be adjusted.